### PR TITLE
Simplify Apollo client by making it web socket-only

### PR DESCRIPTION
**Pull request by rpiaggio**
Piggybacking on this PR: Always copy headers in Fetch JS client. That way, if one request mutates them, it won't interfere with other requests.

### CODE REVIEW

File: core/src/main/scala/clue/websocket/package.scala [hunk 1]
```diff
@@ -3,25 +3,32 @@
...
+import scala.concurrent.duration.FiniteDuration
+
 package object websocket {
-  type WebSocketReconnectionStrategy = ReconnectionStrategy[WebSocketCloseEvent]
+  type CloseEvent = Either[Throwable, CloseParams]
 
-  object WebSocketReconnectionStrategy {
-    def never: WebSocketReconnectionStrategy = ReconnectionStrategy.never
-  }
+  type CloseReason          = Either[Throwable, CloseEvent]
```
Comment (armanbilge) [inline core/src/main/scala/clue/websocket/package.scala:13]: What is the difference between `Right(Left(ex))` and `Left(ex)` ?

**Review by armanbilge — APPROVED** [review]
Thanks, nice refactoring.

File: core/src/main/scala/clue/websocket/package.scala (hunk 1 above)
Comment (rpiaggio) [reply core/src/main/scala/clue/websocket/package.scala]: I added comments in the code. In short, one is a "lost connection" exception, the other one is an "establishing connection" exception.

```scala
  // A Left(exception) here indicates an unexpected/errored disconnection from the server.
  type CloseEvent = Either[Throwable, CloseParams]

  // A Left(exception) here indicates an error establishing the connection.
  type ReconnectReason = Either[Throwable, CloseEvent]

  // Int = Attempt #. Will only be 0 immediately after a close (whether graceful or errored).
  // If the very fist connection attempt fails, the first invocation will have Attempt = 1.
  type ReconnectionStrategy = (Int, ReconnectReason) => Option[FiniteDuration]
```

Comment (rpiaggio) [reply core/src/main/scala/clue/websocket/package.scala]: Also renamed `CloseReason` to the more adequate `ReconnectReason`.

Comment (armanbilge) [reply core/src/main/scala/clue/websocket/package.scala]: Thanks, that helps!

**Review by armanbilge — APPROVED** [review]

**Pull request merged**